### ISSUE TITLE
Avoid over-use of ARIA

### ISSUE THREAD

**Issue by peterh-capella** [formwerkjs/formwerk#117]
I appreciate the effort put into making the forms built with this library accessible. As the [WAI ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/practices/) suggests however, bad ARIA is worse than no ARIA. Where possible, accessibility should be accomplished using non-ARIA methods.

A good example of this is in the "Getting Started" example in the docs. Following those directions and creating a simple text field results in:

`<label id="[generated]" for="[also generated]">text</label>`

and:

`<input type="text" aria-invalid="false" id="[matching for]" value="" aria-labelledby="[matching label id]">`

Since the `for` attribute identifies the label as being connected to the form element with the matching `id` attribute, the `aria-labelledby` attribute on the form element is redundant. Also, the default value for `aria-invalid` is "false" so the attribute is not needed on the input before the form has been submitted. If the default generation of the example was:

`<label id="[generated]" for="[also generated]">text</label>`

and:

`<input type="text" id="[matching for]" value="" >`

It would be a more concise and clear implementation. If additional attributes were required for specific cases, passing those as props similar to how the values for `maxLength` and `required` are provided would be appropriate.

Overall, I would say that this suggestion is of relatively low priority in your development. You've already done a lot of work to make things accessible, so this suggestion would entail refining the approach to remove some things that may not be necessary and could be confusing for some assistive technologies.

**Comment by logaretm** [comment]
> As the [WAI ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/practices/) suggests however, bad ARIA is worse than no ARIA.

I absolutely agree, this is a bug and is not intended.

> Overall, I would say that this suggestion is of relatively low priority in your development.

I think it is relatively easy to fix, Formwerk has a system of capturing refs and is able to tell what kind of elements are present. So we can easily skip adding extra attributes in this case. I believe it worked as intended in the initial version but maybe broke down during refactors.

Thanks for pointing it out and for taking a look! We will fix it ASAP.
